packages/qulacs/quri_parts/qulacs/circuit/noise/gate_converter.py

```diff
@@ -9,36 +9,57 @@
 # limitations under the License.
 
 from collections.abc import Sequence
+from typing import cast
 
 from qulacs import QuantumGateBase
-from qulacs.gate import CPTP, DenseMatrix, Identity, Pauli, Probabilistic
+from qulacs.gate import CPTP, Identity, Pauli, Probabilistic
 
 from quri_parts.circuit.noise import AbstractKrausNoise, PauliNoise, ProbabilisticNoise
+from quri_parts.qulacs import cast_to_list
+from quri_parts.qulacs.circuit import _dense_matrix_gate_qulacs
 
 
 def create_kraus_gate(
     qubits: Sequence[int], noise: AbstractKrausNoise
 ) -> QuantumGateBase:
-    return CPTP([DenseMatrix(qubits, kraus) for kraus in noise.kraus_operators])
+    # This casting is for dealing with missing type of CPTP
+    return cast(
+        QuantumGateBase,
+        CPTP(
+            [
+                _dense_matrix_gate_qulacs(qubits, kraus)
+                for kraus in noise.kraus_operators
+            ]
+        ),
+    )
 
 
 def create_pauli_noise_gate(
     qubits: Sequence[int], noise: PauliNoise
 ) -> QuantumGateBase:
     qubits = noise.qubit_indices
     pauli_list, prob_list = noise.pauli_list, list(noise.prob_list)
+    # have to convert to list(noise.prob_list), otherwise mypy throws:
+    # Argument 1 to "append" of "list" has incompatible type "float"; expected "int"
+    # Even if the `reveal_type(cast_to_list(prob_list))` shows it is a list of floats.
```

Comment:
I don't understand what this comment refers to.
If it is about `list(noise.prob_list)` above, it is because `noise.prob_list` is a tuple and need to be converted into a list.

Reply:
This is explaining why I need to do use `list(noise.prob_list)` on that line instead of doing
`cast_to_list(prob_list).append(prob_identity)` a couple of lines later.

If I do `cast_to_list(prob_list).append(prob_identity)`, `mypy` thinks `cast_to_list(prob_list)` is a list of integers while it is obviously a list of floats, thus throwing the error message `Argument 1 to "append" of "list" has incompatible type "float"; expected "int"`.

I think I'm going to delete the comments and leave the line `list(noise.prob_list)`

Reply:
`list(noise.prob_list)` existed before this PR by a different reason (which I described in the comment).
I'm not sure about the mypy error, but doing `cast_to_list(prob_list).append(...)` instead of `list(prob_list)` will throw in runtime since `prob_list` can be a tuple.

Reply:
I see. I think I confused myself when removing the redundant `list` calls I added.
I have reverted it back.